protocols/identify/src/handler.rs

```diff
@@ -182,9 +182,12 @@ impl Handler {
                         remote_info,
                     )));
             }
-            future::Either::Right(()) => self.events.push(ConnectionHandlerEvent::NotifyBehaviour(
-                Event::IdentificationPushed,
-            )),
+            future::Either::Right(()) => {
+                let local_info = self.build_info();
```

Comment:
Is it guaranteed that this is actually the info we pushed? What if our state changed since we sent it to the other peer?

I'd prefer if we changed the `OutboundUpgrade` implementation such that it emits the `Info` that it sent over the wire so we can carry it forward here.